.github/workflows/doc-tests.yaml

```diff
@@ -142,6 +142,26 @@ jobs:
         working-directory: 'docs'
         run: yarn markdown-lint
 
+      - name: Check out vale
+        uses: actions/checkout@v4
+        with:
+          repository: 'errata-ai/vale'
+          path: 'vale'
+
+      - name: Download vale
+        working-directory: 'vale'
+        env: 
+          GH_TOKEN: ${{ github.token }}
+          VALE_VERSION: 3.12.0
+        run: |
+          gh release download v${VALE_VERSION} --pattern="*Linux_64-bit*"
+          tar -xvf vale_${VALE_VERSION}_Linux_64-bit.tar.gz
```

Comment:
This needs to do a check of the hash so the binaries cannot be changed maliciously. I would also download the specific file in case the wildcard ends up matching more than 1 file later for some reason.

```
VALE_FILENAME="vale_${VALE_VERSION}_Linuc_64-bit.tar.gz"
gh release download -R errata-ai/vale "v${VALE_VERSION}" --pattern="${VALE_FILENAME}"
sha256 --check <<< "3f4ea05cd1f2291b660ecf11a77cbb6b544b0f3b780604ad183f63285611321b  ${VALE_FILENAME}"
tar -xvf "${VALE_FILENAME}"
```

Reply:
Or perhaps add the hash as an env var alongside the version so it is clear(er) they need to be updated together:

```
env:
    VALE_VERSION: 3.12.0
    VALE_SHA256SUM: 3f4ea05cd1f2291b660ecf11a77cbb6b544b0f3b780604ad183f63285611321b
run:
...
sha256sum --check <<<"${VALE_SHA256SUM}  ${VALE_FILENAME}"
```

Reply:
Thanks! I've added the integrity check.